File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Model/Strategy.java
package io.appium.espressoserver.lib.Model;

import io.appium.espressoserver.lib.Exceptions.InvalidStrategyException;

/**
 * Enumerate all possible locator strategies
 */
@SuppressWarnings("unused")
public enum Strategy {
    CLASS_NAME("class name"),
    CSS_SELECTOR("css selector"),
    ID("id"),
    NAME("name"),
    LINK_TEXT("link text"),
    PARTIAL_LINK_TEXT("partial link text"),
    XPATH("xpath"),
    ACCESSIBILITY_ID("accessibility id"),
    TEXT("text"),
    ANDROID_UIAUTOMATOR("-android uiautomator");

    public static Strategy fromString(final String text) throws InvalidStrategyException {
        if (text != null) {
            for (final Strategy s : Strategy.values()) {
                if (text.equalsIgnoreCase(s.strategyName)) {
                    return s;
                }
            }
        }
        throw new InvalidStrategyException("Locator strategy '" + text
                + "' is not supported on Android");
    }

    private final String strategyName;

    Strategy(final String name) {
        strategyName = name;
    }

    public String getStrategyName() {
        return strategyName;
    }
}

File: espresso-server/library/src/main/java/io/appium/espressoserver/lib/model/Strategy.kt
/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * See the NOTICE file distributed with this work for additional
 * information regarding copyright ownership.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.appium.espressoserver.lib.model

import com.google.gson.annotations.SerializedName

/**
 * Enumerate all possible locator strategies
 */
enum class Strategy(val strategyName: String) {
    @SerializedName("class name")
    CLASS_NAME("class name"),
    @SerializedName("css selector")
    CSS_SELECTOR("css selector"),
    @SerializedName("id")
    ID("id"),
    @SerializedName("name")
    NAME("name"),
    @SerializedName("link text")
    LINK_TEXT("link text"),
    @SerializedName("partial link text")
    PARTIAL_LINK_TEXT("partial link text"),
    @SerializedName("xpath")
    XPATH("xpath"),
    @SerializedName("accessibility id")
    ACCESSIBILITY_ID("accessibility id"),
    @SerializedName("text")
    TEXT("text"),
    @SerializedName(value = "-android viewtag", alternate = ["tag name"])
    VIEW_TAG("-android viewtag"),
    @SerializedName(value = "-android datamatcher")
    DATAMATCHER("-android datamatcher"),
    @SerializedName(value = "-android viewmatcher")
    VIEWMATCHER("-android viewmatcher");
}